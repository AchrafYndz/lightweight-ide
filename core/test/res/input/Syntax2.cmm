// Main entry point
fn main() {
    let x = 5;

    if 1 == 1 {
        let y = 5;
    }
    else {
        let x = 5;
    }
    main2();

    while(1 == 1) {
        main3();
    }

    return 0;
}

fn main2() {
    let x = 5;
    let xx = (1 * 1)  /  2;
    let yy  = 1 / 1;
}

fn main3() {
    let xxx = 5;
}