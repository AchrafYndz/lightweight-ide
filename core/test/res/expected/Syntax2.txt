{"function_call_identifier":[{"end":[10,8],"start":[10,4]},{"end":[13,12],"start":[13,8]}],"function_name":[{"end":[1,6],"start":[1,3]},{"end":[19,7],"start":[19,3]},{"end":[25,7],"start":[25,3]}],"identifier":[{"end":[2,8],"start":[2,8]},{"end":[5,12],"start":[5,12]},{"end":[8,12],"start":[8,12]},{"end":[20,8],"start":[20,8]},{"end":[21,9],"start":[21,8]},{"end":[22,9],"start":[22,8]},{"end":[26,10],"start":[26,8]}],"keyword":[{"end":[1,1],"start":[1,0]},{"end":[2,6],"start":[2,4]},{"end":[4,5],"start":[4,4]},{"end":[5,10],"start":[5,8]},{"end":[7,7],"start":[7,4]},{"end":[8,10],"start":[8,8]},{"end":[12,8],"start":[12,4]},{"end":[16,9],"start":[16,4]},{"end":[19,1],"start":[19,0]},{"end":[20,6],"start":[20,4]},{"end":[21,6],"start":[21,4]},{"end":[22,6],"start":[22,4]},{"end":[25,1],"start":[25,0]},{"end":[26,6],"start":[26,4]}],"literal":[{"end":[2,12],"start":[2,12]},{"end":[4,7],"start":[4,7]},{"end":[4,12],"start":[4,12]},{"end":[5,16],"start":[5,16]},{"end":[8,16],"start":[8,16]},{"end":[12,10],"start":[12,10]},{"end":[12,15],"start":[12,15]},{"end":[16,11],"start":[16,11]},{"end":[20,12],"start":[20,12]},{"end":[21,14],"start":[21,14]},{"end":[21,18],"start":[21,18]},{"end":[21,25],"start":[21,25]},{"end":[22,14],"start":[22,14]},{"end":[22,18],"start":[22,18]},{"end":[26,14],"start":[26,14]}]}